### ISSUE TITLE
CycloneDX bom does not include authors/copyright information

### ISSUE THREAD

**Issue by hanna-modica** [oss-review-toolkit/ort#7354]
It looks like, if the authors are set via a curation, they do not appear in the cycloneDX bom as a copyright.

The following curation
```
- id: "Pub:defer_pointer:defer_pointer:0.0.2"
  curations:
    comment: |
     [...]
    concluded_license: "MIT"
    authors:
    - "Copyright (c) 2019 gskinner.com"
```
leads to this in the evaluated model:
```
{
    "_id" : 29,
    "id" : "Pub:defer_pointer:defer_pointer:0.0.2",
    "is_project" : false,
    "definition_file_path" : "",
    "purl" : "pkg:pub/defer_pointer/defer_pointer@0.0.2",
    "authors" : [ "Copyright (c) 2019 gskinner.com" ],
    "declared_licenses_processed" : { },
    "concluded_license" : "MIT",
    "effective_license" : "MIT",
    "description" : "An alternative to Overlay which allows you to easily render and hit test a widget outside its parent bounds.",
    "homepage_url" : "https://github.com/gskinnerTeam/flutter-defer-pointer/",
    "binary_artifact" : {
      "url" : "",
      "hash" : {
        "value" : "",
        "algorithm" : ""
      }
    },
    "source_artifact" : {
      "url" : "",
      "hash" : {
        "value" : "",
        "algorithm" : ""
      }
    },
    "vcs" : {
      "type" : "",
      "url" : "",
      "revision" : "",
      "path" : ""
    },
    "vcs_processed" : {
      "type" : "Git",
      "url" : "https://github.com/gskinnerTeam/flutter-defer-pointer.git",
      "revision" : "",
      "path" : ""
    },
    "curations" : [ {
      "base" : {
        "authors" : [ ]
      },
      "curation" : {
        "comment" : "[...]",
        "authors" : [ "Copyright (c) 2019 gskinner.com" ],
        "concluded_license" : "MIT"
      }
    } ],
    "paths" : [ 32 ],
    "levels" : [ 0 ],
    "scopes" : [ 0 ],
    "is_excluded" : false
  }
```

but in the cycloneDX bom everything from `"curations": ...` seems to be missing:

```
<component type="library">
      <group>defer_pointer</group>
      <name>defer_pointer</name>
      <version>0.0.2</version>
      <description>An alternative to Overlay which allows you to easily render and hit test a widget outside its parent bounds.</description>
      <scope>required</scope>
      <licenses>
        <license>
          <id>MIT</id>
          <text content-type="plain/text" encoding="base64">UGVybWlzc2lvbiBpcyBoZXJlYnkgZ3JhbnRlZCwgZnJlZSBvZiBjaGFyZ2UsIHRvIGFueSBwZXJzb24gb2J0YWluaW5nCmEgY29weSBvZiB0aGlzIHNvZnR3YXJlIGFuZCBhc3NvY2lhdGVkIGRvY3VtZW50YXRpb24gZmlsZXMgKHRoZQoiU29mdHdhcmUiKSwgdG8gZGVhbCBpbiB0aGUgU29mdHdhcmUgd2l0aG91dCByZXN0cmljdGlvbiwgaW5jbHVkaW5nCndpdGhvdXQgbGltaXRhdGlvbiB0aGUgcmlnaHRzIHRvIHVzZSwgY29weSwgbW9kaWZ5LCBtZXJnZSwgcHVibGlzaCwKZGlzdHJpYnV0ZSwgc3VibGljZW5zZSwgYW5kL29yIHNlbGwgY29waWVzIG9mIHRoZSBTb2Z0d2FyZSwgYW5kIHRvCnBlcm1pdCBwZXJzb25zIHRvIHdob20gdGhlIFNvZnR3YXJlIGlzIGZ1cm5pc2hlZCB0byBkbyBzbywgc3ViamVjdCB0bwp0aGUgZm9sbG93aW5nIGNvbmRpdGlvbnM6CgpUaGUgYWJvdmUgY29weXJpZ2h0IG5vdGljZSBhbmQgdGhpcyBwZXJtaXNzaW9uIG5vdGljZSBzaGFsbCBiZQppbmNsdWRlZCBpbiBhbGwgY29waWVzIG9yIHN1YnN0YW50aWFsIHBvcnRpb25zIG9mIHRoZSBTb2Z0d2FyZS4KClRIRSBTT0ZUV0FSRSBJUyBQUk9WSURFRCAiQVMgSVMiLCBXSVRIT1VUIFdBUlJBTlRZIE9GIEFOWSBLSU5ELApFWFBSRVNTIE9SIElNUExJRUQsIElOQ0xVRElORyBCVVQgTk9UIExJTUlURUQgVE8gVEhFIFdBUlJBTlRJRVMgT0YKTUVSQ0hBTlRBQklMSVRZLCBGSVRORVNTIEZPUiBBIFBBUlRJQ1VMQVIgUFVSUE9TRSBBTkQgTk9OSU5GUklOR0VNRU5ULgpJTiBOTyBFVkVOVCBTSEFMTCBUSEUgQVVUSE9SUyBPUiBDT1BZUklHSFQgSE9MREVSUyBCRSBMSUFCTEUgRk9SIEFOWQpDTEFJTSwgREFNQUdFUyBPUiBPVEhFUiBMSUFCSUxJVFksIFdIRVRIRVIgSU4gQU4gQUNUSU9OIE9GIENPTlRSQUNULApUT1JUIE9SIE9USEVSV0lTRSwgQVJJU0lORyBGUk9NLCBPVVQgT0YgT1IgSU4gQ09OTkVDVElPTiBXSVRIIFRIRQpTT0ZUV0FSRSBPUiBUSEUgVVNFIE9SIE9USEVSIERFQUxJTkdTIElOIFRIRSBTT0ZUV0FSRS4K</text><ort:origin xmlns:ort="http://www.w3.org/1999/xhtml">concluded license</ort:origin>
        </license>
      </licenses>
      <purl>pkg:pub/defer_pointer/defer_pointer@0.0.2</purl>
      <modified>false</modified>
      <externalReferences><reference type="website"><url>https://github.com/gskinnerTeam/flutter-defer-pointer/</url></reference></externalReferences><ort:dependencyType xmlns:ort="http://www.w3.org/1999/xhtml">direct</ort:dependencyType>
</component>
```

**Comment by hanna-modica** [comment]
I am not sure, if it is a bug or intended, therefore I will ask here first instead of opening a new issue:
We have another report, which has a NOASSERTION detected. This is overridden in a curation, however, in the cycloneDX bom and the SPDX bom, it still appears as 
```
<license>
  <name>NOASSERTION</name><ort:origin xmlns:ort="http://www.w3.org/1999/xhtml">detected license</ort:origin>
</license>
```
while others, like MIT appear twice (with `concluded license` and `detected license` as `<ort:origin>`)

Is this intended or should concluded licenses should be the only licenses present, like in other reports? If yes, I would open a separate ticket, if it makes sense.

**Comment by sschuberth** [comment]
> Is this intended or should concluded licenses should be the only licenses present, like in other reports?

IIRC, back when I introduced the CycloneDX reporter, I struggled with CycloneDX not having the concept of declared vs detected licenses, i.e. the origin of a license. So I modeled this via an `ExtensibleType` called "origin". For completeness, I thought it would be good to document all origins, even if a concluded license is available, to be able to understand how a license was concluded.

That said, I'd not object to a PR that only keeps a concluded license if present. However, keep in mind that CycloneDX is an SBOM format, and not a report meant for human consumption. I believe the "other reports" you compare to, like probably the web-app report, are meant for human consumption and try to not overwhelm the user with too much information, and thus omit other licenses information if a concluded license if present.

**Comment by hanna-modica** [comment]
Hi @sschuberth,
thank you for explaining and sorry for mixing two topics in one ticket. I answered in #7370. 
Regarding the missing authors from the curation, I am still wondering, why it is missing.

**Comment by sschuberth** [comment]
@hanna-modica can you still reproduce this?

**Comment by franzlst** [comment]
@sschuberth Sorry, I don't have access to our globally defined curations. However, I assume that the curations provided above by @hanna-modica can be used

**Comment by tsteenbe** [comment]
@sschuberth  I was able to reproduce the issue and both CycloneDX and SPDX sbom do not include author data 

Used below curation on [npm-mime-types-2.1.26-scan-result.json.zip](https://github.com/user-attachments/files/17818386/npm-mime-types-2.1.26-scan-result.json.zip).  WebApp report has author change although it's not visualized

```
- id: "NPM::acorn:7.4.1"
  curations:
    comment: |
     Testing authors
    concluded_license: "Apache-2.0"
    authors:
    - "Copyright (c) 2999 Thomas Steenbergen"

```

P.S. In the above i added a concluded license to quickly see if my curation was being picked up.

**Comment by wkl3nk** [comment]
I also currently have a case where the author does not appear in the Disclosure Document.

What is strange: Why do people add this "Copyright (c) ..." to the array of authors? 
From a semantic point of view, an author is a person, so why write something about "Copyright" ?
And is it always the case that the author is the copyright holder?
And what about this global configuration setting 

```kotlin
/**
 * A flag to indicate whether authors should be considered as copyright holders.
 */
val addAuthorsToCopyrights: Boolean = false,
```

**Comment by sschuberth** [comment]
>  I was able to reproduce the issue and both CycloneDX and SPDX sbom do not include author data

To clarify:

1. An [SpdxPackage](https://github.com/oss-review-toolkit/ort/blob/aef875e61ae3ffc0cb2069ddf9eb3778d9722546/utils/spdx/src/main/kotlin/model/SpdxPackage.kt#L29-L33) does not have a literal "author" field like an [ORT Package](https://github.com/oss-review-toolkit/ort/blob/aef875e61ae3ffc0cb2069ddf9eb3778d9722546/model/src/main/kotlin/Package.kt#L60-L65) has. The closest thing are "supplier" / "originator", which haven't been set at all before eead59c4b51e7dd09d2f5294feee1f575f306867 (contained in [ORT release 43.0.0](https://github.com/oss-review-toolkit/ort/releases/tag/43.0.0)).
2. While CycloneDX components do have an "author" field, it was only set for project prior to 097eb5d6397611d0049d090971641c7ea7a128ab (contained in [ORT release 43.0.0](https://github.com/oss-review-toolkit/ort/releases/tag/43.0.0)).

So in the end, I believe the issue to be fixed now, and the "... from curation" part in the original issue title was misleading, as not only authors/copyrights form curations were missing, but any. So the whole issue was unrelated to applying curations.

**Comment by sschuberth** [comment]
> but in the cycloneDX bom everything from `"curations": ...` seems to be missing:

To recap, also here the "everything" was misleading, as actually e.g. the concluded license *is* present. So again, the issue was not related to applying curation data.

**Comment by sschuberth** [comment]
> What is strange: Why do people add this "Copyright (c) ..." to the array of authors?

Ask those people 😉 

> From a semantic point of view, an author is a person, so why write something about "Copyright" ?

Personally, I would agree that a "Copyright" prefix should not be part of any "author" declaration.

> And is it always the case that the author is the copyright holder?

Not always / in all counties / legislations, no.

> And what about this global configuration setting

That's a history crude work-around for the lack of https://github.com/oss-review-toolkit/ort/issues/4519.

**Comment by sschuberth** [comment]
As explained [above](https://github.com/oss-review-toolkit/ort/issues/7354#issuecomment-2535801310), this has been fixed with [ORT release 43.0.0](https://github.com/oss-review-toolkit/ort/releases/tag/43.0.0).
